Automattic/jetpack
# IDC: bring back the IDC screen in Staging mode

This PR partially reverts the changes introduced in #22284.
By disabling IDC in staging mode, we're hiding the fact that staging sites need to setup new connection. This results in staging sites using production blog ID and access tokens, making changes on staging sites affect production.

#### Changes proposed in this Pull Request:
* bring back the proper IDC mode for the sites in staging mode

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?
p9dueE-4ci-p2#comment-7069

#### Testing instructions:
1. Connect the site to Jetpack.
2. Go to "Jetpack Debug -> IDC Simulator" and enable "IDC Simulation", enter the Spoof URL.
3. Click "Send Remote Request", confirm that the site has gone into the IDC mode.
4. Switch the site to the Staging mode: `define( 'JETPACK_STAGING_MODE', true );`
5. Run `wp option delete jetpack_sync_error_idc` (`jetpack docker wp option delete jetpack_sync_error_idc`)
6. Go back to the "IDC Simulator". If you still see the IDC screen, the PR works fine. Otherwise click the "Send Remote Request" and confirm that it shows up.
